kentikapi/internal/api_resources/my_kentik_portal_test.go

```diff
@@ -0,0 +1,228 @@
+	assert.Equal(57, tenants[1].UpdatedDate.Minute())
+	assert.Equal(53, tenants[1].UpdatedDate.Second())
+
+}
+
+func TestGetTenantInfo(t *testing.T) {
+	getTenantInfoResponse := `
+	{
+		"id": 577,
+		"name": "test_tenant",
+		"description": "This is test tenant",
+		"users": [
+			{
+				"id": "148099",
+				"user_email": "test@tenant.user",
+				"last_login": null,
+				"tenant_id": "577",
+				"company_id": "74333"
+			},
+			{
+				"id": "148113",
+				"user_email": "user@testtenant.com",
+				"last_login": null,
+				"tenant_id": "577",
+				"company_id": "74333"
+			}
+		],
+		"created_date": "2020-12-21T10:55:52.449Z",
+		"updated_date": "2020-12-21T10:55:52.449Z"
+	}`
+
+	transport := &api_connection.StubTransport{ResponseBody: getTenantInfoResponse}
+	myKentikPortalAPI := api_resources.NewMyKentikPortalAPI(transport)
+
+	//act
+	tenant, err := myKentikPortalAPI.Get(nil, 577)
+
+	//assert
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	assert.Zero(transport.RequestBody)
+
+	assert.Equal(models.ID(577), tenant.ID)
+	assert.Equal("test_tenant", tenant.Name)
+	assert.Equal("This is test tenant", tenant.Description)
+	assert.Len(tenant.Users, 2)
+	assert.Equal(2020, tenant.CreatedDate.Year())
+	assert.Equal(time.Month(12), tenant.CreatedDate.Month())
+	assert.Equal(21, tenant.CreatedDate.Day())
+	assert.Equal(10, tenant.CreatedDate.Hour())
+	assert.Equal(55, tenant.CreatedDate.Minute())
+	assert.Equal(52, tenant.CreatedDate.Second())
+	assert.Equal(2020, tenant.UpdatedDate.Year())
+	assert.Equal(time.Month(12), tenant.UpdatedDate.Month())
+	assert.Equal(21, tenant.UpdatedDate.Day())
+	assert.Equal(10, tenant.UpdatedDate.Hour())
+	assert.Equal(55, tenant.UpdatedDate.Minute())
+	assert.Equal(52, tenant.UpdatedDate.Second())
+	assert.Equal(models.ID(148099), tenant.Users[0].ID)
+	assert.Equal("test@tenant.user", tenant.Users[0].Email)
+	assert.Nil(tenant.Users[0].LastLogin)
+	assert.Equal(models.ID(577), tenant.Users[0].TenantID)
+	assert.Equal(models.ID(74333), tenant.Users[0].CompanyID)
+	assert.Equal(models.ID(148113), tenant.Users[1].ID)
+	assert.Equal("user@testtenant.com", tenant.Users[1].Email)
+	assert.Nil(tenant.Users[1].LastLogin)
+	assert.Equal(models.ID(577), tenant.Users[1].TenantID)
+	assert.Equal(models.ID(74333), tenant.Users[1].CompanyID)
+}
+
+func TestTenantUserCreate(t *testing.T) {
+	createTenantUserResponse := `
+	{
+		"id": "158564",
+		"user_email": "test@test.test",
+		"last_login": null,
+		"tenant_id": "578",
+		"company_id": "74333"
+	}`
+
+	transport := &api_connection.StubTransport{ResponseBody: createTenantUserResponse}
+	myKentikPortalAPI := api_resources.NewMyKentikPortalAPI(transport)
+
+	//act
+	tenantUser, err := myKentikPortalAPI.CreateTenantUser(nil, 577, "test@test.test")
+
+	//assert
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	payload := utils.NewJSONPayloadInspector(t, transport.RequestBody)
```

Comment:
I think that we should check that API does not send redundant data, e.g.
```json
{
  "user": {
    "user_email": "foo",
    "redundant_trash": "trash",
  }
}
```

I suggest to check number of keys in JSON objects.

Reply:
API does return more fields than we deserialize. They are described as "internal-use" in the doc and we consciously ignore them.

Reply:
I meant the request payload that the library sends (API client, not API).

Reply:
@lukaa12 Please add a TODO for that, e.g.
`TODO: Verify that that there is no redundant data sent in request body`

It is important to check that we do not send harmful data that would e.g. zero value of a field - set it to `""` or `0`.

Reply:
TODO added